atomic_reactor/schemas/container.json

```diff
@@ -4,229 +4,376 @@
   "title": "container image build configuration",
 
   "type": ["object", "null"],
-  "properties": {
-    "go": {
-      "type": "object",
-      "properties": {
-        "modules": {
-          "type": ["array", "null"],
-          "items": {
-            "type": "object",
-            "properties": {
-              "module": {
-                "type": "string",
-                "description": "Top-level Go module (package) name which will be built",
-                "examples": [
-                  "example.com/go/packagename",
-                  "example.com/go/anotherpackage"
-                ]
+  "anyOf": [
+      {
+         "properties": {
+            "go": {
+              "type": "object",
+              "properties": {
+                "modules": {
+                  "type": ["array", "null"],
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "module": {
+                        "type": "string",
+                        "description": "Top-level Go module (package) name which will be built",
+                        "examples": [
+                          "example.com/go/packagename",
+                          "example.com/go/anotherpackage"
+                        ]
+                      },
+                      "archive": {
+                        "type": "string",
+                        "description": "Possibly-compressed archive containing full source code including dependencies",
+                        "examples": [
+                          "anotherpackage.tar.gz"
+                        ]
+                      },
+                      "path": {
+                        "type": "string",
+                        "description": "Path to directory containing source code (or its parent), possibly within archive",
+                        "examples": [
+                          "anotherpackage-v0.54.1"
+                        ]
+                      }
+                    },
+                    "additionalProperties": false,
+                    "required": ["module"]
+                  }
+                }
               },
-              "archive": {
-                "type": "string",
-                "description": "Possibly-compressed archive containing full source code including dependencies",
-                "examples": [
-                  "anotherpackage.tar.gz"
-                ]
+              "additionalProperties": false
+            },
+            "platforms": {
+              "type": ["object", "null"],
```

Comment:
Would be nicer to put these other property schema into definitions and reference them in each alternative.

Reply:
sure